terraform/domains/environment_domains/main.tf

```diff
@@ -1,15 +1,33 @@
+# module "domains" {
+#   source              = "./vendor/modules/domains//domains/environment_domains"
+#   zone                = var.zone
+#   front_door_name     = var.front_door_name
+#   resource_group_name = var.resource_group_name
+#   domains             = var.domains
+#   environment         = var.environment_short
+#   host_name           = var.origin_hostname
+#   null_host_header    = var.null_host_header
+# }
+# module "records" {
+#   source      = "./vendor/modules/domains//dns/records"
+#   hosted_zone = var.hosted_zone
+
+# }
+
 module "domains" {
-  source              = "./vendor/modules/domains//domains/environment_domains"
-  zone                = var.zone
-  front_door_name     = var.front_door_name
-  resource_group_name = var.resource_group_name
-  domains             = var.domains
-  environment         = var.environment_short
-  host_name           = var.origin_hostname
-  null_host_header    = var.null_host_header
+  for_each              = var.hosted_zone
+  source                = "git::https://github.com/DFE-Digital/terraform-modules.git//domains/environment_domains?ref=stable"
+  zone                  = each.key
+  front_door_name       = each.value.front_door_name
+  resource_group_name   = each.value.resource_group_name
+  domains               = each.value.domains
+  environment           = each.value.environment_short
+  host_name             = each.value.origin_hostname
+  null_host_header      = try(each.value.null_host_header, false)
```

Comment:
null_host_header is always set to false here, and the default is false: https://github.com/DFE-Digital/terraform-modules/blob/main/domains/environment_domains/variables.tf#L17
So it can be deleted from here